app/src/examples/RuntimeTests/tests/core/useSharedValue.test.tsx

```diff
@@ -0,0 +1,249 @@
+/* eslint-disable no-inline-styles/no-inline-styles */
+import React, { useEffect } from 'react';
+import { View } from 'react-native';
+import { useSharedValue } from 'react-native-reanimated';
+import {
+  describe,
+  test,
+  expect,
+  render,
+  registerValue,
+  getRegisteredValue,
+  Presets,
+  clearRenderOutput,
+} from '../../ReanimatedRuntimeTestsRunner/RuntimeTestsApi';
+import { ComparisonMode } from '../../ReanimatedRuntimeTestsRunner/types';
+
+const SHARED_VALUE_REF = 'SHARED_VALUE_REF';
+const DOUBLED_SHARED_VALUE_REF = 'SHARED_VALUE_REF';
+
+const ProgressBar = ({ progress }: { progress: number }) => {
+  return (
+    <View style={{ margin: 30, borderWidth: 3, borderColor: 'darkorange' }}>
+      <View
+        style={{
+          height: 80,
+          width: `${100 * progress}%`,
+          backgroundColor: 'darkorange',
+        }}
+      />
+    </View>
+  );
+};
+
+describe('Tests of sharedValue', () => {
+  describe('Test setting different values as sharedValue', () => {
+    const SharedValueComponent = ({ initialValue, progress }: { initialValue: unknown; progress: number }) => {
+      const sharedValue = useSharedValue(initialValue);
+      registerValue(SHARED_VALUE_REF, sharedValue);
+      return <ProgressBar progress={progress} />;
+    };
+    test.each([
+      { presetName: 'numbers', comparisonMode: ComparisonMode.NUMBER },
+      { presetName: 'bigInts', comparisonMode: ComparisonMode.NUMBER },
+      { presetName: 'serializableArrays', comparisonMode: ComparisonMode.ARRAY },
+      { presetName: 'strings', comparisonMode: ComparisonMode.STRING },
+      { presetName: 'serializableObjects', comparisonMode: ComparisonMode.OBJECT },
```

Comment:
Do you use all Presets here?

Reply:
No, I don't use maps, sets, unserializable arrays (buffer arrays) etc.